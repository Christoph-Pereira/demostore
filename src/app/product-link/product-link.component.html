<a routerLink="{{url}}">
    <div class="productLinkDiv" [ngStyle]="{backgroundImage: 'url(' + image + ')'}">
        <p class="productLinkText">
            {{text}}
        </p>
    </div>
</a>