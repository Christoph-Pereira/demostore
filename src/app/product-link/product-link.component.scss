a {
    text-decoration: none;
}
.productLinkDiv {
    height: 450px;
    width: 685px;
    margin: auto;
    transition: all .2s ease-in-out;

    .productLinkText {
        font-size: 2rem;
        padding: 2rem;
        color: white;
    }
}

.productLinkDiv:hover {
    transform: scale(1.1);
}