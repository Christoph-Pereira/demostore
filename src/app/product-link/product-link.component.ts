import { Component, Input, OnInit } from '@angular/core';

@Component({
  selector: 'app-product-link',
  templateUrl: './product-link.component.html',
  styleUrls: ['./product-link.component.scss']
})
export class ProductLinkComponent {
  
  @Input() public text: String | undefined;
  @Input() public image: String | undefined;
  @Input() public url: String | undefined;

  constructor() { }

}
