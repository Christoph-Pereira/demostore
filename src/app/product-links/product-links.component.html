<div class="productLinksDiv">
    <h1>Showcased Solutions</h1>
    <div class ="productLinks">
        <div *ngFor="let productLink of productLinks">
            <app-product-link [text]="productLink.text" [image]="productLink.image" [url]="productLink.url"></app-product-link>
        </div>
    </div>
</div>