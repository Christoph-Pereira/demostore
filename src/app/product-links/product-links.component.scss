.productLinksDiv {
    align-items: center;
    color: white;
    padding: 2rem;

    h1 {
        text-align: center;
        color: black;
        margin: 2rem;
        font-size: 2rem;

    }
    .productLinks {
        display: grid;
        grid-template-columns: auto auto;
    }
}