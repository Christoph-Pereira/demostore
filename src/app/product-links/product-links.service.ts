import { Injectable } from '@angular/core';

@Injectable({
  providedIn: 'root'
})
export class ProductLinksService {

  constructor() { }

  
  private productLinks = [
    {
      "text": "Advanced Gamification",
      "image": "../../assets/images/Centrical-background-2.png",
      "url": "/product/1"
    },
    {
      "text": "Integrated Occupational Health",
      "image": "../../assets/images/Sodales-background-1-.png",
      "url": "/product/2"
    }
  ]

  getProductLinks() {
    return this.productLinks;
  }
}
