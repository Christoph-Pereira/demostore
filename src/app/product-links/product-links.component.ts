import { Component, OnInit } from '@angular/core';
import { ProductService } from '../api/products.service';
import { ProductLinksService } from './product-links.service';
import { first, Subscription } from 'rxjs';

interface productLink {
  text: string,
  image: string,
  url: string
}
@Component({
  selector: 'app-product-links',
  templateUrl: './product-links.component.html',
  styleUrls: ['./product-links.component.scss']
})
export class ProductLinksComponent implements OnInit {

  public productLinks: productLink[] | undefined;
  public randomProductTitleText = "";
  private randomProductTitleSubscription: Subscription = new Subscription;

  constructor(private productLinksService: ProductLinksService, private productService: ProductService) { }

  ngOnInit(): void {
    this.productLinks = this.productLinksService.getProductLinks();

    this.randomProductTitleSubscription = this.productService.getRandomProductTitle()
    .subscribe(randomProductTitleText => this.randomProductTitleText = randomProductTitleText);
  }

  ngOnDestroy(): void {
    this.randomProductTitleSubscription.unsubscribe();
  }

}
