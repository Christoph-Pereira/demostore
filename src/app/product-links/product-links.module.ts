import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ProductLinkComponent } from '../product-link/product-link.component';
import { ProductLinksComponent } from './product-links.component';
import { RouterModule } from '@angular/router';



@NgModule({
  declarations: [ProductLinkComponent, ProductLinksComponent],
  imports: [
    CommonModule,
    RouterModule
  ],
  exports: [ProductLinksComponent]
})
export class ProductLinksModule { }
