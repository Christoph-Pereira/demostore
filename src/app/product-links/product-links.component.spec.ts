import { ComponentFixture, TestBed } from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { MockComponent } from 'ng-mocks';
import { ProductLinkComponent } from '../product-link/product-link.component';

import { ProductLinksComponent } from './product-links.component';
import { ProductLinksService } from './product-links.service';

describe('ProductLinksComponent', () => {
  let component: ProductLinksComponent;
  let fixture: ComponentFixture<ProductLinksComponent>;
  let productLinks = [{
    "text": "Advanced Gamification",
    "image": "../../assets/images/Centrical-background-2.png",
    "url": "/product/1"
  },
  {
    "text": "Integrated Occupational Health",
    "image": "../../assets/images/Sodales-background-1-.png",
    "url": "/product/2"
  }];
  const mockedProductLinksService: any = {getProductLinks: jest.fn().mockReturnValue(
    productLinks)};

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [ ProductLinksComponent, MockComponent(ProductLinkComponent) ],
      providers: [
        { provide: ProductLinksService, useValue: mockedProductLinksService }
      ]
    })
    .compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ProductLinksComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should contain productLinks', () => {
    expect(component.productLinks).toBeTruthy();
  });
  
  it('should match snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should have 2 productLinks', () => {
    const appProductLinks = fixture.nativeElement.querySelectorAll('app-product-link');
    expect(appProductLinks.length).toEqual(2);
  });
  
  it('should have 3 productLinks', () => {
    productLinks.push({
      "text": "Integrated Occupational Health",
      "image": "../../assets/images/Sodales-background-1-.png",
      "url": "/product/2"
    });
    fixture.detectChanges();
    const appProductLinks = fixture.nativeElement.querySelectorAll('app-product-link');
    expect(appProductLinks.length).toEqual(3);
  });

  it('should pass values to productLink', () => {
    const productLink: ProductLinkComponent = fixture.debugElement.query(By.css('app-product-link')).componentInstance;
    expect(productLink.text).toBeTruthy();
    expect(productLink.image).toBeTruthy();
    expect(productLink.url).toBeTruthy();
  });
});
