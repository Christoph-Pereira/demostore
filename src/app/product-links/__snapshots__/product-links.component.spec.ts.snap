// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductLinksComponent should match snapshot 1`] = `
<app-product-links
  __ngContext__={[Function LRootView]}
  productLinks={[Function Array]}
  productLinksService={[Function Object]}
>
  <div
    class="productLinksDiv"
  >
    <h1>
      Showcased Solutions
    </h1>
    <div
      class="productLinks"
    >
      <div>
        <app-product-link />
      </div>
      <div>
        <app-product-link />
      </div>
    </div>
  </div>
</app-product-links>
`;
